server/controllers/pages.js

```diff
@@ -16,7 +16,27 @@ const entryHelper = require('../helpers/entry')
  * Edit document in Markdown
  */
 router.get('/edit/*', (req, res, next) => {
-  if (!res.locals.rights.write) {
+  var xPaths = ['news', 'general'] // the final word in the url For.e.g : news in /edit/news, general in /edit/general 
+
+  var checkEditAvailable = function (xPaths) {
+    let paths = req.path.split('/')
+
+    let path = paths[ paths.length - 1 ]
+
+    for (let i = 0; i < xPaths.length; i++) {
+      if (path === xPaths[i]) {
+        return false
+      }
+    }
+    return true
+  }
+
+  let rt_array = req.user.rights
+  let rt_array_len = req.user.rights.length
+
+  let role = rt_array[rt_array_len - 1].role
```

Comment:
I'm not sure why ypu are reading the last element of the rights array, but is there any chance that the user is an admin but we wouldn't be able to find that? Check out Array.find() as that'd be apt here (pass it a function to check if  the currently being checked element has the admin role).

(Read docs on MDN).

Reply:
I am reading the last element because the wiki mongo data  had admin role in the 2nd element in the array. 
I didn't generalize it though.

Reply:
I'm thinking that adding any custom roles would change the ordering. Still, better to be generic than debugging what went wrong later.